### PR TITLE
Fix the path for users which have an exception for a group share

### PR DESCRIPTION
**Pull request by nickvergessen**
### Steps to reproduce
1. User1 is member of Group1
2. User1 creates folder test
3. User2 creates folder test
4. User2 shares folder test with Group1
5. User2 creates a file in test
6. User1 has test and the shared "test (2)"
### Expected

User1 has activity "User2 created `Tooltip: in test (2)` test.txt
### Actual

User1 has activity "User2 created `Tooltip: in test` test.txt

Activity app is the only code using the `$returnUserPaths === true` method.

@rullzer @schiesbn

### CODE REVIEW

**Comment by nickvergessen** [comment]
Changeset without whitespaces: https://github.com/owncloud/core/pull/17330/files?w=1

File: lib/private/share/constants.php [hunk 1]
```diff
@@ -27,6 +27,7 @@ class Constants {
 
 	const SHARE_TYPE_USER = 0;
 	const SHARE_TYPE_GROUP = 1;
+	const SHARE_TYPE_GROUP_USER = 2;
```
Comment (schiesbn) [inline lib/private/share/constants.php]: It is already defined here https://github.com/owncloud/core/pull/17330/files#diff-7a65827e967104a42d0097681fed3cdaR30

The reasons: This share type is only used internally and nothing which should be exposed to the users of the share API

Comment (nickvergessen) [inline lib/private/share/constants.php]: fixed

**Comment by nickvergessen** [comment]
@karlitschek please approve backport for 8.1.1

**Comment by schiesbn** [comment]
looks good :+1:

**Comment by PVince81** [comment]
:+1: works

**Comment by PVince81** [comment]
This somehow reminds me of this https://github.com/owncloud/core/pull/15961
I only wrote unit tests but didn't have time for the fix.
Is this related ?

**Comment by nickvergessen** [comment]
:+1: Yes, thanks for the tests, made me aware of the "owner renames the folder, but is also in the group".

Fixed the case and all your tests pass now.

**Comment by PVince81** [comment]
Works :+1:

**Comment by schiesbn** [comment]
looks good :+1:

**Comment by MorrisJobke** [comment]
@karlitschek Backport was requested here https://github.com/owncloud/core/pull/17330#issuecomment-118036105 ;)

**Comment by karlitschek** [comment]
:-) Please backport :-)

**Comment by nickvergessen** [comment]
Backport: https://github.com/owncloud/core/pull/17653

**Pull request merged**